mh/pat.py

```diff
@@ -1759,15 +1761,38 @@ def sendNtcFriendAdd(self, recipient_id, accepted, seq):
         """
         if not accepted:
             return
-        data = b""
-        data += pati.lp2_string(recipient_id)
+
+        users = self.session.find_users(recipient_id, "", 1, 1)
+        assert len(users)>0, "User is offline or does not exist."
```

Comment:
This sounds like a bad idea, a user accepting a friend invite shouldn't be disconnected because the other user is offline or doesn't exist. It should have not received the friend invite in the first place.

Reply:
If a user sends a friend request and then goes offline before it's accepted, what should be the proper behavior? Should it still allow the other party to accept it, and then the offline user will see the new addition to their friends list when they log back in? This would be a fairly simple thing to write, I just want to make sure that that's the intended behavior.

Reply:
When the user is not online or doesn't exist you can reply with a [AnsAlert](https://github.com/sepalani/MH3SP/blob/dev/mh/pat.py#L145) with the proper message to notify the player of the situation.

Reply:
Done